src/native/corehost/hostmisc/pal.windows.cpp

```diff
@@ -292,18 +332,26 @@ bool pal::get_default_installation_dir(pal::string_t* recv)
     }
 
     append_path(recv, _X("dotnet"));
-    if (pal::is_emulating_x64())
+    if (is_current_arch && pal::is_emulating_x64())
     {
         // Install location for emulated x64 should be %ProgramFiles%\dotnet\x64.
-        append_path(recv, _X("x64"));
+        append_path(recv, get_arch_name(arch));
+    }
+#if defined(TARGET_ARM64)
+    else if (!is_current_arch)
+    {
+        // Running arm64, looking for x64 install
+        assert(arch == pal::known_architecture::x64);
+        append_path(recv, get_arch_name(arch));
     }
+#endif
 
     return true;
 }
 
 namespace
 {
-    void get_dotnet_install_location_registry_path(HKEY * key_hive, pal::string_t * sub_key, const pal::char_t ** value)
+    void get_dotnet_install_location_registry_path(pal::known_architecture arch, HKEY * key_hive, pal::string_t * sub_key, const pal::char_t ** value)
```

Comment:
I'm somewhat conflicted about this. I kind of see the output as showing state of the world, so if somebody added a registry key / config file to register a custom location for any other arch for whatever reason, I feel like we should show it. Similarly, this is currently dumping all possible DOTNET_ROOT environment variables. (For the default path case, I put the 'supported' check since the default path is the same for different architectures without explicit support.)

Reply:
I'm also not sure - so feel free to keep as is. It was just a difference I noticed, but it's as you say.